src/charm.py

```diff
@@ -316,6 +362,11 @@ def _handle_status_update_config(self, event: HookEvent) -> None:
             logger.info("Hydra service is absent. Deferring the event.")
             return
 
+        if not self._log_level_valid:
+            logger.info(f"Invalid configuration value for log_level: {self._log_level}")
+            self.unit.status = BlockedStatus("Bad configuration value for log_level")
+            return
+
```

Comment:
we could check if the log level value is valid before rendering the template - it shouldn't be rendered with a wrong config value